### PR TITLE
Live Notification updates in notification fragment

### PR DESCRIPTION
**Pull request by VivekUchiha**

### CODE REVIEW

File: app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt [hunk 1]
```diff
@@ -144,11 +150,26 @@ class NotificationFragment : Fragment() {
         }
     }
 
+    private val refreshNotifications = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            customNotificationList.add(Notification(intent.getStringExtra("text")!!, intent.getStringExtra("createdat")!!))
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: Use string constants, define in this fragment itself in companion object. Also avoid !! null check

File: app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt [hunk 2]
```diff
@@ -144,13 +150,37 @@ class NotificationFragment : Fragment() {
...
+        super.onResume()
+        //IF IT IS ALIVE DUE TO SOME MISHAP IT IS UNREGISTERED
+        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(refreshNotifications)
+        val intentFilter = IntentFilter(MainActivity.REFRESH_UNREAD_NOTIFICATIONS_COUNT)
+        LocalBroadcastManager.getInstance(requireContext()).registerReceiver(refreshNotifications, IntentFilter(intentFilter))
+    }
+
     override fun onDestroy() {
         super.onDestroy()
+        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(refreshNotifications)
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: In one of the projects I had run into this. Maybe, that's why you have to unregister in onResume()? Basically use onPause()?
https://stackoverflow.com/questions/17195641/fragment-lifecycle-when-ondestroy-and-ondestroyview-are-not-called

File: app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt [hunk 3]
```diff
@@ -144,13 +150,37 @@ class NotificationFragment : Fragment() {
...
+
     override fun onPause() {
         super.onPause()
+        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(refreshNotifications)
         preferences.edit().remove(Constants.LAST_NOTIFICATION_ID).apply()
     }
 
+    override fun onResume() {
+        super.onResume()
+        //IF IT IS ALIVE DUE TO SOME MISHAP IT IS UNREGISTERED
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: Can you elaborate on the "Mishap" ?

File: app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt (hunk 2 above)
Comment (VivekUchiha) [reply app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: I am using onPause too. It still doesn't seem to unregister at times.

Comment (VivekUchiha) [reply app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: I've fixed it. The problem was that I was registering once in onView created too so this + onResume registered it twice.

Comment (Harsh2098) [reply app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: COol

Comment (Harsh2098) [reply app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: So remove this from onDestroy() now?

Comment (VivekUchiha) [reply app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: It is still needed in onDestroy. The problem was it was getting Registered twice at the start.

File: app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt [hunk 4]
```diff
@@ -144,13 +148,34 @@ class NotificationFragment : Fragment() {
...
     override fun onPause() {
         super.onPause()
+        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(refreshNotifications)
         preferences.edit().remove(Constants.LAST_NOTIFICATION_ID).apply()
     }
 
+    override fun onResume() {
+        super.onResume()
+        val intentFilter = IntentFilter(MainActivity.REFRESH_UNREAD_NOTIFICATIONS_COUNT)
+        LocalBroadcastManager.getInstance(requireContext()).registerReceiver(refreshNotifications, IntentFilter(intentFilter))
```
Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt]: Replace `IntentFilter(intentFilter)` with just `intentFilter` ?

**Comment by Harsh2098** [comment]
Resolve conflicts and squash if required

**Pull request merged**